src/mimeparser.rs

```diff
@@ -340,6 +334,17 @@ impl<'a> MimeParser<'a> {
         self.header.contains_key("chat-version")
     }
 
+    pub(crate) fn get_subject(&self) -> Option<String> {
+        if let Some(s) = self.header.get("subject") {
```

Comment:
More functional way to write this: `self.header.get("subject").filter(|s| !s.is_empty()).map(|s| s.to_string())` (have not checked if it compiles though, maybe needs some tweaks)